mm2src/mm2_main/tests/docker_tests/docker_tests_inner.rs

```diff
@@ -955,8 +962,308 @@ fn test_watcher_spends_taker_spends_maker_payment() {
     assert_eq!(bob_mycoin_balance, "97.99999");
 }
 
+#[test]
+fn test_watcher_waits_for_taker() {
+    let (_ctx, _, bob_priv_key) = generate_utxo_coin_with_random_privkey("MYCOIN", 100.into());
+    generate_utxo_coin_with_privkey("MYCOIN1", 100.into(), bob_priv_key);
+    let (_ctx, _, alice_priv_key) = generate_utxo_coin_with_random_privkey("MYCOIN1", 100.into());
+    generate_utxo_coin_with_privkey("MYCOIN", 100.into(), alice_priv_key);
+    let watcher_priv_key = *SecretKey::new(&mut rand6::thread_rng()).as_ref();
+
+    let coins = json!([
+        {"coin":"MYCOIN","asset":"MYCOIN","txversion":4,"overwintered":1,"txfee":1000,"protocol":{"type":"UTXO"}},
+        {"coin":"MYCOIN1","asset":"MYCOIN1","txversion":4,"overwintered":1,"txfee":1000,"protocol":{"type":"UTXO"}},
+    ]);
```

Comment:
Could you please add `my_coin_conf` and `my_coin1_conf` helpers? Better late than never 🙂